subsys/bluetooth/host/classic/sdp.c

```diff
@@ -1475,29 +1496,39 @@ void bt_sdp_init(void)
 
 int bt_sdp_register_service(struct bt_sdp_record *service)
 {
-	uint32_t handle = SDP_SERVICE_HANDLE_BASE;
+	uint8_t index = 0;
 
 	if (!service) {
 		LOG_ERR("No service record specified");
 		return 0;
 	}
 
-	if (num_services == BT_SDP_MAX_SERVICES) {
-		LOG_ERR("Reached max allowed registrations");
-		return -ENOMEM;
+	if (sys_slist_find(&sdp_db, &service->node, NULL)) {
+		LOG_ERR("Service already registered");
+		return -EEXIST;
 	}
 
-	if (db) {
-		handle = db->handle + 1;
+	if (!sys_slist_is_empty(&sdp_db)) {
```

Comment:
when call bt_sdp_register_service exceed UINT8_MAX, it would always return -EOVERFLOW. If we unregister  node, even if there is only one node, whoes index is 254,  we could not register any more.